dolphinscheduler-ui/src/views/datasource/list/use-form.ts

```diff
@@ -269,6 +269,15 @@ export function useForm(id?: number) {
     } else {
       state.showPrincipal = false
     }
+
+    state.showRestEndpoint = false
+    state.showNamespace = false
+    state.showKubeConfig = false
+    state.showAccessKeyId = false
+    state.showAccessKeySecret = false
+    state.showRegionId = false
+    state.showEndpoint = false
```

Comment:
It's better put these into each type's else logic so that it can be better maintained.

Reply:
> It's better put these into each type's else logic so that it can be better maintained.

done